airflow/api_fastapi/core_api/routes/public/dags.py

```diff
@@ -303,3 +307,45 @@ def delete_dag(
             status.HTTP_409_CONFLICT, f"Task instances of dag with id: '{dag_id}' are still running"
         )
     return Response(status_code=status.HTTP_204_NO_CONTENT)
+
+
+@dags_router.get(
+    "/{dag_id}/assets/queuedEvent",
+    responses=create_openapi_http_exception_doc(
+        [
+            status.HTTP_401_UNAUTHORIZED,
+            status.HTTP_403_FORBIDDEN,
+            status.HTTP_404_NOT_FOUND,
+        ]
+    ),
+)
+def get_dag_asset_queued_events(
+    dag_id: str,
+    session: Annotated[Session, Depends(get_session)],
+    before: str = Query(None),
+) -> QueuedEventCollectionResponse:
+    """Get queued asset events for a DAG."""
+    where_clause = [AssetDagRunQueue.target_dag_id == dag_id]
+    if before:
+        before_parsed = timezone.parse(before)
+        where_clause.append(AssetDagRunQueue.created_at < before_parsed)
+    query = (
+        select(AssetDagRunQueue, AssetModel.uri)
+        .join(AssetModel, AssetDagRunQueue.asset_id == AssetModel.id)
+        .where(*where_clause)
+    )
+    result = session.execute(query).all()
+    total_entries = len(result)
+    if not result:
+        raise HTTPException(status.HTTP_404_NOT_FOUND, f"Queue event with dag_id: `{dag_id}` was not found")
```

Comment:
I think you removed the `uri` from the error response message. I think it is important to keep it for the client to understand which event was not found.

Reply:
Nope, it was never there. The reason it is not there is because we are getting "ALL" the asset events. So it doesnt make sense to show that. That would make sense if we got one asset event by "uri"

Reply:
Reference:
![image](https://github.com/user-attachments/assets/e17a5079-5151-4e4a-a06b-5a2ebce0fc3a)